core/src/main/java/org/wso2/testgrid/tinkerer/TinkererSDK.java

```diff
@@ -0,0 +1,232 @@
+                authenticationString.getBytes(StandardCharsets.UTF_8));
+        if (this.tinkererHost == null) {
+            logger.warn("Tinkerer host does not initialized properly");
+        }
+    }
+
+    /**
+     * Send shell command as http request to the Tinkerer and get back the result as asynced result.
+     *
+     * @param agentId   id of the agent to send command
+     * @param command   command to execute
+     * @return      Response handler as Async response
+     */
+    public AsyncCommandResponse executeCommandAsync(String agentId, String command) {
+        Client client = ClientBuilder.newClient();
+        String operationId = UUID.randomUUID().toString();
+        OperationRequest operationRequest = new OperationRequest(command,
+                OperationRequest.OperationCode.SHELL, agentId);
+        operationRequest.setOperationId(operationId);
+        Gson gson = new Gson();
+        String jsonRequest = gson.toJson(operationRequest);
+        logger.debug("Sending async commands to " + this.tinkererHost  + " agent " + agentId);
+        Response response = client.target(this.tinkererHost)
+                .path("stream-operation")
+                .request()
+                .header(HttpHeaders.AUTHORIZATION, this.authenticationToken)
+                .post(Entity.entity(jsonRequest,
+                        MediaType.APPLICATION_JSON));
+        Path filePath = Paths.get(System.getProperty("java.io.tmpdir"));
+        ScriptExecutorThread scriptExecutorThread = new ScriptExecutorThread(operationRequest.getOperationId()
+                , response, filePath);
+        scriptExecutorThread.start();
+        AsyncCommandResponse asyncCommandResponse = new AsyncCommandResponse(operationId,
+                filePath, scriptExecutorThread);
+        asyncCommandResponse.setOperationId(operationRequest.getOperationId());
+        return asyncCommandResponse;
+    }
+
+    /**
+     * Send shell command as http request to the Tinkerer and get back the result as synced response.
+     *
+     * @param agentId   id of the agent to send command
+     * @param testPlanId    The test plan id
+     * @param instantName   The instant name
+     * @param command   command to execute
+     * @return  Response handler as Sync response
+     */
+    public SyncCommandResponse executeCommandSync(String agentId, String testPlanId, String instantName,
+                                                  String command) {
+        Client client = ClientBuilder.newClient();
+        String operationId = UUID.randomUUID().toString();
+        OperationRequest operationRequest = new OperationRequest(command,
+                OperationRequest.OperationCode.SHELL, agentId);
+        operationRequest.setOperationId(operationId);
+        Gson gson = new Gson();
+        String jsonRequest = gson.toJson(operationRequest);
+        logger.debug("Sending sync commands to " + this.tinkererHost + " agent " + agentId);
```

Comment:
Remove this debug log or check if the debug logging is enabled prior to logging.